locator/token_metadata.cc

```diff
@@ -101,27 +99,31 @@ class token_metadata_impl final {
     token_metadata_impl(const token_metadata_impl&) = delete; // it's too huge for direct copy, use clone_async()
     token_metadata_impl(token_metadata_impl&&) noexcept = default;
     const std::vector<token>& sorted_tokens() const;
-    future<> update_normal_tokens(std::unordered_set<token> tokens, inet_address endpoint);
+    future<> update_normal_tokens(std::unordered_set<token> tokens, host_id endpoint);
     const token& first_token(const token& start) const;
     size_t first_token_index(const token& start) const;
-    std::optional<inet_address> get_endpoint(const token& token) const;
-    std::vector<token> get_tokens(const inet_address& addr) const;
-    const std::unordered_map<token, inet_address>& get_token_to_endpoint() const {
+    std::optional<host_id> get_endpoint(const token& token) const;
+    std::vector<token> get_tokens(const host_id& addr) const;
+    const std::unordered_map<token, host_id>& get_token_to_endpoint() const {
         return _token_to_endpoint_map;
     }
 
-    const std::unordered_set<inet_address>& get_leaving_endpoints() const {
+    const std::unordered_set<host_id>& get_leaving_endpoints() const {
         return _leaving_endpoints;
     }
 
-    const std::unordered_map<token, inet_address>& get_bootstrap_tokens() const {
+    const std::unordered_map<token, host_id>& get_bootstrap_tokens() const {
         return _bootstrap_tokens;
     }
 
     void update_topology(inet_address ep, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> shard_count = std::nullopt) {
```

Comment:
Can this function be deleted at the end?

Reply:
Yes, already did in the new PR version.

Reply:
It's still here.